legate/core/launcher.py

```diff
@@ -916,7 +912,7 @@ def build_single_task(self, argbuf: BufferBuilder) -> SingleTask:
 
     def execute(
         self, launch_domain: Rect, redop: Optional[int] = None
-    ) -> Union[Future, FutureMap]:
+    ) -> FutureMap:
```

Comment:
If we want to do this, then we should also remove `redop` and the if branch in the body to make sure there's no code that can ever produce `Future`.

Reply:
Weird, my comment here got lost. I had asked if this needed to be split, or if there is another solution. The reason for the change is that `Reduce` does this:
```
            weights = launcher.execute(launch_domain)
            ...
            weighted = Weighted(self._runtime, launch_shape, weights)
```
but `Weighted` does expect a `Future`
> legate/core/operation.py:787:62: error: Argument 3 to "Weighted" has incompatible type "Union[Future, FutureMap]"; expected "FutureMap"  [arg-type]

Reply:
I think there might not actually be anywhere that passed `redop` though, in which case just removing it will be simple

Reply:
https://github.com/nv-legate/legate.core/pull/195/commits/7e9c5d3c73c79c9e92e74296a25e65a766aa2216